src/act.obj.cpp

```diff
@@ -3674,11 +3674,8 @@ void perform_remove(struct char_data * ch, int pos)
   if (GET_OBJ_TYPE(obj) == ITEM_HOLSTER)
     GET_HOLSTER_READY_STATUS(obj) = 0;
   /* add damage back from stim patches */
-  /* it doesn't do anything to keep track, */
-  /* so I'm just makeing it a mod mental damage to it */
   else if ( GET_OBJ_TYPE(obj) == ITEM_PATCH && GET_PATCH_TYPE(obj) == PATCH_STIM ) {
-    GET_MENTAL(ch) = GET_PATCH_STIMPATCH_ORIGINAL_MENTAL(obj);
-    GET_PATCH_STIMPATCH_ORIGINAL_MENTAL(obj) = 0;
+    GET_MENTAL(ch) = MAX(0, GET_MENTAL(ch) - (GET_PATCH_RATING(patch) * 100));
```

Comment:
What was the issue with the old code? I'm not sure why it was written the way it was, but I expect it was something to do with preventing underflow issues (which this change does as well)

Reply:
Ah, you looked at it in the minute after I committed and before I could comment about it. The problem was that it's exploitable with predictable sources of stun. If you know stun damage is coming, you patch right before it happens and then remove the patch right after it happened.

Reply:
Makes sense, appreciate the patch then.